OctopusDSC/DSCResources/cOctopusWorkerPool/cOctopusWorkerPool.psm1

```diff
@@ -0,0 +1,248 @@
+        WorkerPoolName        = $WorkerPoolName
+        WorkerPoolDescription = $WorkerPoolDescription
+        SpaceId               = $SpaceId
+        OctopusCredentials    = $OctopusCredentials
+        OctopusApiKey         = $OctopusApiKey
+    }
+
+    return $result
+}
+
+function Set-TargetResource {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSDSCUseVerboseMessageInDSCResource", "")]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Url,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Present", "Absent")]
+        [string]$Ensure,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$WorkerPoolName,
+        [string]$WorkerPoolDescription,
+        [string]$SpaceId,
+        [PSCredential]$OctopusCredentials = [PSCredential]::Empty,
+        [PSCredential]$OctopusApiKey = [PSCredential]::Empty
+    )
+
+    $currentResource = Get-TargetResource -Url $Url `
+        -Ensure $Ensure `
+        -WorkerPoolName $WorkerPoolName `
+        -OctopusCredentials $OctopusCredentials `
+        -OctopusApiKey $OctopusApiKey
+
+    if ($Ensure -eq "Absent" -and $currentResource.Ensure -eq "Present") {
+        Remove-WorkerPool -Url $Url `
+            -WorkerPoolName $WorkerPoolName `
+            -OctopusCredentials $OctopusCredentials `
+            -OctopusApiKey $OctopusApiKey
+    }
+    elseif ($Ensure -eq "Present" -and $currentResource.Ensure -eq "Absent") {
+        New-WorkerPool -Url $Url `
+            -WorkerPoolName $WorkerPoolName `
+            -WorkerPoolDescription $WorkerPoolDescription `
+            -SpaceId $SpaceId `
+            -OctopusCredentials $OctopusCredentials `
+            -OctopusApiKey $OctopusApiKey
+
+    }
+}
+
+function Test-TargetResource {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSDSCUseVerboseMessageInDSCResource", "")]
+    [OutputType([boolean])]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Url,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Present", "Absent")]
+        [string]$Ensure,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$WorkerPoolName,
+        [string]$WorkerPoolDescription,
+        [string]$SpaceId,
+        [PSCredential]$OctopusCredentials = [PSCredential]::Empty,
+        [PSCredential]$OctopusApiKey = [PSCredential]::Empty
+    )
+    $currentResource = (Get-TargetResource -Url $Url `
+            -Ensure $Ensure `
+            -WorkerPoolName $WorkerPoolName `
+            -WorkerPoolDescription $WorkerPoolDescription `
+            -SpaceID $SpaceId `
+            -OctopusCredentials $OctopusCredentials `
+            -OctopusApiKey $OctopusApiKey)
+
+    $params = Get-ODSCParameter $MyInvocation.MyCommand.Parameters
+
+    $currentConfigurationMatchesRequestedConfiguration = $true
```

Comment:
Can we use a common implementation of this in the helper file?

Reply:
We probably can now that multiple resources rely on it yeah. definitely on the list.